sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/GlobalEndpointManager.java

```diff
@@ -159,6 +166,20 @@ public void close() {
         });
     }
 
+    public Mono<DatabaseAccount> getDatabaseAccountFromCache(URI defaultEndpoint) {
+        return this.databaseAccountAsyncCache.getAsync(StringUtils.EMPTY, null, () -> this.owner.getDatabaseAccountFromEndpoint(defaultEndpoint).single().doOnSuccess(databaseAccount -> {
+            if(databaseAccount != null) {
+                this.latestDatabaseAccount = databaseAccount;
+            }
+
+            this.refreshLocationAsync(databaseAccount, false);
+        }));
+    }
+
+    public DatabaseAccount getLatestDatabaseAccount() {
+        return this.latestDatabaseAccount;
```

Comment:
there is a time window where GlobalEndpointManager is instantiated but hasn't fetched DatabaseAccount, within that period how do we ensure this never returns null?

Reply:
We are calling init on GlobalEndpointManager  from RxDocumentClientImpl which will update this value , any future call if return null due to any exception or error we will not update latestDatabaseAccount and async cahce , they will hold the previous not null value

Reply:
Please document the new API saying what you said above.

Reply:
done